### ISSUE TITLE
[Question] Support for YI Dome X

### ISSUE THREAD

**Issue by adicrescenzo** [roleoroleo/yi-hack-MStar#262]
Hi all,

I bought the model YI Dome X on [amazon](https://www.amazon.it/gp/product/B07QZRJ98M/ref=ppx_yo_dt_b_asin_title_o06_s00?ie=UTF8&psc=1).

![](https://m.media-amazon.com/images/S/aplus-media/sc/e392d2c1-5683-42de-9721-86dd65834a02.__CR0,0,300,300_PT0_SX300_V1___.png)

Those are the values reported above the box of the camera: 

DID: BFUSY02....
SN: BY30EU....
Model: YYS.3017

I tried to flash the firmware y203c_0.3.6 but nothing happened to the camera, the default firmware is still working.
The firmware version in the Yi Home app is 4.0.0.0F_201909251646

I thought that the DID value is the one to identify the camera model, is that correct? Is this camera supported for the hack? Many thanks in advance.

**Comment by roleoroleo** [comment]
DID is not enough.
Based on fw version your cam is not supported.

**Comment by adicrescenzo** [comment]
> 
> 
> DID is not enough.
> Based on fw version your cam is not supported.

Is the camera model correct for this hack? Are there any chance for the future to support this firmware version?
Thanks

**Comment by roleoroleo** [comment]
I don't know.
To check if it's possible to create an hack we need to dump the memory and analyze it.

**Comment by adicrescenzo** [comment]
How can I make it? I would really like to contribute

**Comment by roleoroleo** [comment]
See the wiki.

**Comment by adicrescenzo** [comment]
Done! Can you give some hints to find the 3 uart pads?

I took some pictures of the circuite board:

Back
![image](https://user-images.githubusercontent.com/6871290/96840527-db0b0500-144a-11eb-8e70-7f85fad06f5b.jpeg)

Front
![image](https://user-images.githubusercontent.com/6871290/96840559-e827f400-144a-11eb-9606-f29eb0d4a8fe.jpeg)

Front from another angle
![image](https://user-images.githubusercontent.com/6871290/96840586-f249f280-144a-11eb-88d4-fb97e064b41d.jpeg)

**Comment by roleoroleo** [comment]
The chip is MStar but I can't see uart pads.

**Comment by Roihu** [comment]
UART pads (circled red) can be found under the speaker module, which is attached to the PCB with two screws (circled green). Unfortunately I do not myself have access to an UART adapter right now, so I cannot dump firmware yet.
![DomeX](https://user-images.githubusercontent.com/1498824/97084362-08101100-161f-11eb-8b65-8c344569d761.jpg)

**Comment by adicrescenzo** [comment]
> UART pads (circled red) can be found under the speaker module, which is attached to the PCB with two screws (circled green). Will post the dump soon if successful.
> ![DomeX](https://user-images.githubusercontent.com/1498824/97084362-08101100-161f-11eb-8b65-8c344569d761.jpg)

That’s a great news! Thank you very much for the hint. Tomorrow I’ll disassemble the speaker and I’ll try the make a memory dump with the uart interface.

**Comment by Roihu** [comment]
I managed to find my own UART adapter and have started the dump, the process seems to be extremely slow, (rootfs seems like it will take around 2 hours, which would mean home will take around 12 hours as it is six times larger). I suppose increasing the --step argument in the dump script would speed this up, but is there a risk of getting corrupted data then? The code for the dump script suggests a default step size of 256, while the wiki says 64 (which I'm using now)

**Comment by roleoroleo** [comment]
The uart you are using hasn't control signals (RTS and CTS).
If you use big blocks data will be corrupted.

**Comment by Roihu** [comment]
> The uart you are using hasn't control signals (RTS and CTS).
> If you use big blocks data will be corrupted.

I see, thank you. Guess I'll just leave it running overnight. 
Attached is the swapped sys_y30 file. I will update this message to include the home_y30 file tomorrow once the dump is finished. 
[sys_y30.swapped.img.zip](https://github.com/roleoroleo/yi-hack-MStar/files/5433429/sys_y30.swapped.img.zip)

**Comment by TheTimeWalker** [comment]
Any news about this? I can spend a beer as an incentive if this can be hacked

**Comment by roleoroleo** [comment]
Please try this:
[y30_0.3.8.tar.gz](https://github.com/roleoroleo/yi-hack-MStar/files/5532237/y30_0.3.8.tar.gz)

But make a backup copy before you try.

**Comment by Roihu** [comment]
Awesome, flashed fine and at least for now I didn't find anything that _didn't_ work. ONVIF (including PTZ), RTSP, SSH, Telnet, all seems to work. Thank you!

**Comment by roleoroleo** [comment]
Great!

**Comment by TheTimeWalker** [comment]
Keeping my promise, a beer here costs 8 CHF, so I took that as the donation price

![image](https://user-images.githubusercontent.com/8660693/99147880-9e66ae00-2684-11eb-8e54-63bb87f55dec.png)

**Comment by roleoroleo** [comment]
Thank you for your donation!

**Comment by adicrescenzo** [comment]
> Awesome, flashed fine and at least for now I didn't find anything that _didn't_ work. ONVIF (including PTZ), RTSP, SSH, Telnet, all seems to work. Thank you!

Great news! Did you flash it copying the files on the sdcard?

**Comment by wasserpanther** [comment]
Flashed by extracting the tar.gz provided onto the sd card. Got all Camera functions working (PTZ, RTSP, etc.).
One small issue: It seems that the camera looses the WiFi credentials after reboot, I used the "save config" option to see if they got saved in some of the config files but i could not find any information. Am I missing something there or could this be related to bug #66 ?
Thanks for your work, donation is on its way :+1:

**Comment by roleoroleo** [comment]
It's a very old problem that affects all platforms, but only a few users.
I have never been able to solve it

The "Save config" doesn't save wifi informations.
You can save them only through the yi app.

Thank you for your donation!

**Comment by WestLight-dev** [comment]
Hi all,

one question: are home_y30 and home_y30m two different cameras? Or can I rename the file home_y30 to homey3m?

Thank you!

**Comment by roleoroleo** [comment]
home_y30m and home_y30 are related to the same cam, but with different file formats.
home_y30m is a file created to upgrade a file system, it's compressed bz2 or 7z and crypted and it's processed by a linux script.
home_y30 is a file containing a file system image created to overwrite completely a partition, it's "compressed" squash or jffs2 and it's processed by u-boot.

**Comment by WestLight-dev** [comment]
I copied the two files (home_y30 and sys_y30) to the memory card but after rebooting the camera it did not work :-(

**Comment by adicrescenzo** [comment]
Hi guys, I tried to make a copy of the firmware with the UART adapter but when I type the commands listed in the wiki like "sf probe", nothing appends. I've used a mac book pro with 115200 ad baud rate. Do you have any suggestions? 

Flashing the new firmware I had no problems at all, everything works as expected. Great work @roleoroleo, my donation is on its way too.

Last question: is it possibile to start the recording with a web service invocation?

**Comment by roleoroleo** [comment]
> Hi guys, I tried to make a copy of the firmware with the UART adapter but when I type the commands listed in the wiki like "sf probe", nothing appends. I've used a mac book pro with 115200 ad baud rate. Do you have any suggestions?

No suggestions, try to continue.

> Flashing the new firmware I had no problems at all, everything works as expected. Great work @roleoroleo, my donation is on its way too.

Thank you for your donation.

> Last question: is it possibile to start the recording with a web service invocation?

Like after a motion detection event?

**Comment by adicrescenzo** [comment]
> 
> 
> > Hi guys, I tried to make a copy of the firmware with the UART adapter but when I type the commands listed in the wiki like "sf probe", nothing appends. I've used a mac book pro with 115200 ad baud rate. Do you have any suggestions?
> 
> No suggestions, try to continue.
> 
Ok I'll do it. If the command takes effect is there any feedback from the camera?
In the guide published on the wiki there is a point "Stop the boot at the U-Boot prompt (should be "MStar #") pressing "Enter"."
Even if I press enter the boot process continues, is it normal? 

> > Flashing the new firmware I had no problems at all, everything works as expected. Great work @roleoroleo, my donation is on its way too.
> 
> Thank you for your donation.
> 
> > Last question: is it possibile to start the recording with a web service invocation?
> 
> Like after a motion detection event?

No, I would like to start the recording manually invoking a web service. In the same way we can move the camera left or right.
Thanks.

**Comment by roleoroleo** [comment]
> In the guide published on the wiki there is a point "Stop the boot at the U-Boot prompt (should be "MStar #") pressing "Enter"."
> Even if I press enter the boot process continues, is it normal?

No, check if the TX pin is connected correctly.

> No, I would like to start the recording manually invoking a web service. In the same way we can move the camera left or right.
> Thanks.

Currently is not possible to start a video recording but probably I could try to write a new ws that send a motion detection event to the system. The yi software should start a new record.

**Comment by roleoroleo** [comment]
https://github.com/roleoroleo/yi-hack-MStar/commit/f4504af6369bba8085353069dc9e95da50c665b4

Try these:
[ipc_cmd.gz](https://github.com/roleoroleo/yi-hack-MStar/files/5549039/ipc_cmd.gz)
[record.sh.gz](https://github.com/roleoroleo/yi-hack-MStar/files/5549040/record.sh.gz)

**Comment by adicrescenzo** [comment]
> [f4504af](https://github.com/roleoroleo/yi-hack-MStar/commit/f4504af6369bba8085353069dc9e95da50c665b4)
> 
> Try these:
> [ipc_cmd.gz](https://github.com/roleoroleo/yi-hack-MStar/files/5549039/ipc_cmd.gz)
> [record.sh.gz](https://github.com/roleoroleo/yi-hack-MStar/files/5549040/record.sh.gz)
> 
> The url is http://IP_CAM:8080/cgi-bin/record.sh?time=30

Great it works!! I've just see a little problem, the invocation to the web service is synchronous and the http call doesn't terminate until the time parameter is reached (e.g 30 or 60 seconds). Is there a way to make this call asynchronous?

Thanks

**Comment by roleoroleo** [comment]
Yes, I could fork the process.

**Comment by WestLight-dev** [comment]
I need your help! Please! I have a YI Dome X (6FUS) with 4.0.0.0 firmware ...I have tried everything, but nothing helped. I cannot overwrite the firmware.

**Comment by roleoroleo** [comment]
Does the camera work normally when you remove the sd card?

**Comment by WestLight-dev** [comment]
@roleoroleo 
Yes, it does. I have two other cameras (YI Outdoor 1080p) ... both of them worked immediately.

**Comment by WestLight-dev** [comment]
After switching on the camera (with sd card into) the yellow light goes on ... for about 20 seconds. Then the camera goes to the right and back again. Afterwards the light will glow blue.

**Comment by roleoroleo** [comment]
Try to change the sd card.

**Comment by WestLight-dev** [comment]
I tried with 2GB, 16GB and 64GB ... FAT32 formatted ... very strange :-(

**Comment by roleoroleo** [comment]
What the firmware versione?
I need all digits.

**Comment by adicrescenzo** [comment]
> 
> 
> @adicrescenzo
> [922affb](https://github.com/roleoroleo/yi-hack-MStar/commit/922affb375d1420db6f07f7dc2577365cf4cca80)

Thank you very much! How can I get the new compiled firmware? I'll try to start the development myself but first of all I need to understand how it works :)

**Comment by roleoroleo** [comment]
If you don't want to wait the next release, try to build it yourself.
See the wiki.

**Comment by thoemmes86** [comment]
Good Morning,

and many many thanks for offering the support for this camera.
I was waiting a very long time to get this camera supported. So the offering of a beer is for sure.
Both Dome X cameras where able to install the firmware and work like a charm.

BUT:
At least I found one Bug which is only available on one of those identical Dome X Devices.
For ONLY ONE of them, the 'High res snapshot' is not working while the 'Low res snapshot' and all other functionalities are.

The configurations of both cameras are identical, I tried a reset of the camera, but nothing solved the issue.
You may have an idea how to solve?

Kind Regards

**Comment by WestLight-dev** [comment]
@roleoroleo : current firmware is **4.0.0.0F_201909251646**

**Comment by roleoroleo** [comment]
@thoemmes86 
Is the yi fw version the same?

@WestLight-dev 
The fw versione is ok, it's supported.
Check please if sd is correctly detected from the cam (with yi app).
Are you able to watch the videos recorded from the cam?

**Comment by thoemmes86** [comment]
@roleoroleo 
Yes it is exactly the same.

**Comment by roleoroleo** [comment]
> @roleoroleo 
> Yes it is exactly the same.

Does high res stream work?

**Comment by WestLight-dev** [comment]
sd-card is detected correctly. I can see the logs and videos on the sd-card.

**Comment by WestLight-dev** [comment]
![WhatsApp Image 2020-11-18 at 10 59 45](https://user-images.githubusercontent.com/74542190/99515946-eab83380-298d-11eb-86ad-849914f3b225.jpeg)

**Comment by thoemmes86** [comment]
> > @roleoroleo
> > Yes it is exactly the same.
> 
> Does high res stream work?

Yes everything else is working as expected. (As far as I have tested)

**Comment by roleoroleo** [comment]
@thoemmes86 
Try to run the command manually.
`imggrabber -r HIGH -w -d > /tmp/sd/snap.jpg`
and post the output.

**Comment by thoemmes86** [comment]
imggrabber -r HIGH -w -d > /tmp/sd/snap.jpg
watermark on
debug on
Resolution: 1080
vaddr: 0xb4745000 - paddr: 0x23570000 - size: 368640
copy buffer: len 368640
copy again buffer: len 368640
Segmentation fault

**Comment by thoemmes86** [comment]
So I have identified that there is also an issue with the high res stream which was not that obvious.
RTSP Stream is configured as HIGH.
It is possible to open the RTSP stream, but you only get a low resolution stream.

**Comment by roleoroleo** [comment]
It seems the same problem I already encountered on allwinner platform.
Probably it's a different model.
Please make a backup copy of your firmware using dd and share it with me.

**Comment by thoemmes86** [comment]
May I'm outing myself as a noob, but currently I was not able to do a DD backup via SSH and storing it on the internal SD.
Do you have some commands I can use for it.

**Comment by roleoroleo** [comment]
Login with ssh and run the following commands:
https://github.com/roleoroleo/yi-hack-MStar/issues/223#issuecomment-722469856

Repeat for each mtdblock.

**Comment by thoemmes86** [comment]
That is what I tried.

 # cat /tmp/sd/backup/mtd.txt
dev:    size   erasesize  name
mtd0: 00050000 00010000 "BOOT"
mtd1: 00180000 00010000 "KERNEL"
mtd2: 001e0000 00010000 "ROOTFS"
mtd3: 00c30000 00010000 "HOME"
mtd4: 00010000 00010000 "vd1"
mtd5: 00010000 00010000 "conf"
/ # dd if=/dev/mtd0 of=/tmp/sd/backup/mtd0.bin
dd: can't open '/dev/mtd0': No such file or directory
/ # dd if=/dev/mtdblock0 of=/tmp/sd/backup/mtdblock0.bin
dd: can't open '/dev/mtdblock0': No such file or directory

**Comment by roleoroleo** [comment]
Try:

```
cat /proc/mtd > /tmp/sd/backup/mtd.txt

dd if=/dev/mtd/mtd0 of=/tmp/sd/backup/mtdblock0.bin
dd if=/dev/mtd/mtd1 of=/tmp/sd/backup/mtdblock1.bin
dd if=/dev/mtd/mtd2 of=/tmp/sd/backup/mtdblock2.bin
...
```

**Comment by thoemmes86** [comment]
Worked, thanks.
[backup.zip](https://github.com/roleoroleo/yi-hack-MStar/files/5565301/backup.zip)

**Comment by roleoroleo** [comment]
Your hack is ok. I don't understand.

**Comment by thoemmes86** [comment]
Hmmm... do you think may a camera hard reset could help or will this lead to potential issues with the hack?

**Comment by thoemmes86** [comment]
That's it... a reset with the reset Button at the bottom and a re-pair with the app solved the issue.
Thanks for your time. Could you confirm the deletion of the backup?
As I don't know how good the encryption of yi is... :-D

**Comment by roleoroleo** [comment]
> Thanks for your time. Could you confirm the deletion of the backup?
> As I don't know how good the encryption of yi is... :-D

Yes.

**Comment by thoemmes86** [comment]
I invested several hours this weekend to test the hack at the dome x and it is working very smooth with my two cameras.

During this test, I faced some incompatibility or may a missing functionality in the motion detection.

The dome x supports human detection with a specific setting in the yi app. This is a very nice feature because it reduces the number of wrong alerts.

The issue is, as soon as you activate this setting, the mqtt activity trigger is not longer working and sends no more activity data.
If you deactivate human detection it is again working.

**Comment by roleoroleo** [comment]
It's a know issue, also present on other platforms (allwinner for example).
At the moment I don't know how to solve it.

**Comment by TheTimeWalker** [comment]
My Dome X finally arrived and I can confirm the hack installation works perfectly. Quick testing, I can use PTZ to move around the camera on the web interface and I can watch HQ stream without issues.

**Comment by WestLight-dev** [comment]
How can i reconnect the camera, after hard reset (Button at the bottom)

**Comment by roleoroleo** [comment]
Repeat the pairing process with yi app.
But you need "Cloud disable" off.

**Comment by roleoroleo** [comment]
Released in 0.3.9.

**Comment by roleoroleo** [comment]
Thank you for your support.
I close the issue.

**Comment by fizzi** [comment]
i bought a yi dome x with firmware 8.1.0.0A_2020012111401. I don't find any information about this firmware. Any suggestion?

**Comment by antodeilippo** [comment]
New Firmware 8.1.0.0A_2020012111401 also for my new Yi Dome Camera.  I tried to use y30_0.4.1.tgz but it doesn't work, Camera start but ignore file on SD Card.

I'm ready with another couple of beers :-)

**Comment by roleoroleo** [comment]
Please try to create a backup with the procedure explained in the wiki of allwinner-v2 model.
And share it with me.

**Comment by fizzi** [comment]
Ok i will, thanks

**Comment by antodeilippo** [comment]
[Here ](https://drive.google.com/file/d/13JXGY7amRX1DETBFG5g7HyK-oWv3Dk7g/view?usp=sharing) you can find the file generated with backup procedure explained in allwinner-v2 model.

**Comment by fizzi** [comment]
I have already shared my backup with @roleoroleo

**Comment by roleoroleo** [comment]
Your model is allwinner based (suffix y30qa).
Please open a new issue here https://github.com/roleoroleo/yi-hack-Allwinner

**Comment by vin-w** [comment]
@Roihu and @adicrescenzo 

I am trying to open my Yi Dome  #368
The screw at the back of the camera under the speaker is removed.

<a href="https://imgbb.com/"><img src="https://i.ibb.co/5sTFTGz/IMG-4100.jpg" alt="IMG-4100" border="0"></a><br />

<a href="https://ibb.co/6vMg0Dn"><img src="https://i.ibb.co/pW7Jj40/IMG-4101.jpg" alt="IMG-4101" border="0"></a>

What will be the next step?
Your help is much appreciated.

**Comment by adicrescenzo** [comment]
> @Roihu and @adicrescenzo
> 
> I am trying to open my Yi Dome #368
> The screw at the back of the camera under the speaker is removed.
> 
> [![IMG-4100](https://camo.githubusercontent.com/b310f6288e5c4ac5b96344819559334801ab7aa932228b9d9a746ec9b62403db/68747470733a2f2f692e6962622e636f2f3573544654477a2f494d472d343130302e6a7067)](https://imgbb.com/)
> 
> [![IMG-4101](https://camo.githubusercontent.com/85209ae10baac559d51d1e2e925c5f5c24d3a6259fa53a53d03831b19ba384de/68747470733a2f2f692e6962622e636f2f7057374a6a34302f494d472d343130312e6a7067)](https://ibb.co/6vMg0Dn)
> 
> What will be the next step?
> Your help is much appreciated.

You need to separate the body of the camera (colored gray) from the white cover pulling down the grey part:

[IMG-4101](https://i.ibb.co/q5THq2m/IMG-4101.jpg)

I suggest to use an opener tool like this:

[IMG-4384](https://i.ibb.co/vmkS5WY/IMG-4384.jpg)

it is useful to separate the grey part from the white cover before pulling it down. Don't be afraid to use a little force :)

**Comment by vin-w** [comment]
<a href="https://ibb.co/LRKSPpm"><img src="https://i.ibb.co/yP20Rst/IMG-4170.jpg" alt="IMG-4170" border="0" width=300></a>
@adicrescenzo  thanks for your help. Finally opened and also find UART.
@roleoroleo I think this is MStar too ?

<a href="https://ibb.co/Jd7x54m"><img src="https://i.ibb.co/MhV8M4Z/IMG-4169.jpg" alt="IMG-4169" border="0" width=300>></a>
I tried to use CH340 to connect serial .
But when I start the camera, it is dumping strange characters.
Is CH340 ok with yi cam?

```
# this is the command I called on mac after installed the driver
screen /dev/cu.usbserial-141140
```

**Comment by vin-w** [comment]
@adicrescenzo after you opened the cam, did you able to revert it back to original state?
The motor wheel is blocking me from plugging in the white cover.

<a href="https://ibb.co/D4K4dpD"><img src="https://i.ibb.co/dBKBZg0/IMG-4445.jpg" alt="IMG-4445" border="0" width="500"></a>

**Comment by adicrescenzo** [comment]
@vin-w sure, I've opened the camera many times and every time I was able to revert it back. 
I'm sorry but I can't understand what is going wrong in the picture you posted.

**Comment by vin-w** [comment]
@adicrescenzo thanks for your update!

My problem is.. in the motor there is a black motor gear.
After I used my force to open the white cover, 
when I want to put it back..(circle in blue)
the gear was blocked by the edge due to not enough size (see the red edge highlighted)

Hope it makes easier to understand

<a href="https://ibb.co/fdcZfR3"><img src="https://i.ibb.co/HTJR6vc/IMG-4450.jpg" alt="IMG-4450" border="0" width="500"></a>
<a href="https://ibb.co/QCV4MK6"><img src="https://i.ibb.co/z7gzZ4S/IMG-4452.jpg" alt="IMG-4452" border="0" width="500"></a><br />

**Comment by Stephonomon** [comment]
Confirming this still works in 2026, and adding the detail that sends people to the
wrong project.

Installed **y30 0.5.7** on a Yi Dome X, and everything works: RTSP (h264 1920x1088
@20fps + pcm_mulaw 8kHz audio), snapshot, ONVIF, PTZ pan/tilt, MQTT, SSH.

```
model     YYS.3017
serial    BFUSY02...          <- matches the DID in the first post here
firmware  4.0.0.0F_201909251646
status.json: name=yi-hack-mstar fw_version=0.5.7 model_suffix=y30 ptz=yes
```

**The model number alone is not enough to choose a project.** YYS.3017 exists on two
different platforms, and the *firmware version* (and serial prefix) is the
discriminator:

| Model | Serial prefix | Firmware | Project |
| --- | --- | --- | --- |
| YYS.3017 | `BFUSY02` | `4.0.0.0F_*` | **yi-hack-MStar `y30`** (this issue) |
| YYS.3017 | `BFUSY29` | `8.1.0.0A_*` | yi-hack-Allwinner `y30qa` (see #356 and Allwinner-v2 #1093, where @roleoroleo answers "Check Allwinner project: y30qa") |

This may be worth adding to the README: the MStar "Supported cameras" table has no
serial-prefix column, but the "Is my cam supported?" section says to check "the
serial number (first 4 letters) and the firmware version... if both numbers appear
in the same row in the table above". For the YI Dome Camera X row there is no serial
to match against, so BFUSY02 vs BFUSY29 is currently undocumented.

Also worth noting for anyone landing here: `status.json` reports
`serial_number: shdnqEv67dC000000000` on this camera, which is a placeholder, not the
real serial. Use the label on the device.

Thanks to @PyryW for the original UART dump and @roleoroleo for the build - still
running six years later.

---
*Investigated and drafted with [Claude Code](https://claude.com/claude-code).*

**Comment by roleoroleo** [comment]
I added a link to this issue in the readme.
